apiv2/internal/server/utils.go

```diff
@@ -62,6 +66,15 @@ func SafeInt32ToUint32(n int32) (uint32, error) {
 	return uint32(n), nil
 }
 
+// SafeUint32ToUint64 converts an uint32 to uint64 safely.
+func SafeUint32ToUint64(n uint32) (uint64, error) {
+	res := uint64(n)
+	if uint32(res) != n { //nolint:gocritic,gosec // no risk of overflow
+		return 0, errors.New("uint32 wrongly converted to uint64")
```

Comment:
should we use our errors pkg here? The same comment applies below.

Reply:
Done.